### PR TITLE
refactor: remove async usage from bucket

### PR DESCRIPTION
**Pull request by callmehiphop**
Relates to #746

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by callmehiphop** [comment]
Working on tests, but early feedback would be lovely!

File: samples/bucketLock.js [hunk 1]
```diff
@@ -34,7 +34,9 @@ async function setRetentionPolicy(bucketName, retentionPeriod) {
     .bucket(bucketName)
     .setRetentionPeriod(retentionPeriod);
   console.log(
-    `Bucket ${bucketName} retention period set for ${metadata.retentionPolicy.retentionPeriod} seconds.`
+    `Bucket ${bucketName} retention period set for ${
```
Comment (JustinBeckwith) [inline samples/bucketLock.js]: Going to want to `rm -rf node_modules package-lock.json` && `npm run fix` here.  Prettier made an update in a patch release I think that's causing some grief.

Comment (callmehiphop) [reply samples/bucketLock.js]: Whoops, well this is embarrassing 😱

File: src/bucket.ts [hunk 2]
```diff
@@ -30,11 +30,12 @@ import {
 import {paginator} from '@google-cloud/paginator';
 import {promisifyAll} from '@google-cloud/promisify';
 import arrify = require('arrify');
-import * as async from 'async';
 import * as extend from 'extend';
 import * as fs from 'fs';
 import * as mime from 'mime-types';
 import * as path from 'path';
+import pLimit from 'p-limit';
```
Comment (bcoe) [inline src/bucket.ts]: have we opted to move away from `async` to `p-limit` for other libraries. Curious what the advantages are.

Comment (callmehiphop) [reply src/bucket.ts]: Good question, this is the only one that I'm aware of. I saw that #731 planned on pulling it in as a regular dependency, so I'm sort of piggy backing off of that.

AFAIK most of the ways we used `async` here can be replaced by native Promises, but managing concurrency is not something that comes out of the box there. In the past I've usually turned to [p-queue](https://www.npmjs.com/package/p-queue), but `p-limit` works here too and we were already slated to use it.

Comment (AVaksman) [reply src/bucket.ts]: Followed the suggestion [here](https://github.com/googleapis/nodejs-spanner/pull/449#issuecomment-442529440)

**Pull request merged**